### ISSUE TITLE
DOS'ing self when turning phone on again after much texting with desktop app

### ISSUE THREAD

**Issue by mallyvai** [WhisperSystems/Signal-Android#6126]
I have:
- [x] searched open and closed issues for duplicates
- [x] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

----------------------------------------

### Bug description
App floods phone with nonstop notifications after desktop texting

### Steps to reproduce
- Turn off your phone
- Open the Signal Private Messenger app
- Do a bunch of texting with the desktop app. 
- Make sure to view all messages sent back at you.
- Turn on the phone

**Actual result:** Watch your phone flood with notifications - buzzing, etc. It will last for many minutes. In my case, almost 5 full minutes. My phone died, I needed to do a bunch of texting via the webapp, and when I turned the app on again - I basically DOS'ed my own phone. It was going through messages slowly and individually, since it gets the message first, sends the notification, and only then marks its read/unread state.

**Expected result:** 
From a user experience/product perspective, if I was opening the app after the phone was shutdown (or reception was reentered or whatever), I wouldn't want to see the buzzing, and have to wait. I'd simply see a single notification (or perhaps, no notification at all), and be able to use my phone immediately.

tl;dr I shouldn't get a notification at all for a message if it was read on either platform.

### Screenshots
(Not a single event, so no screenshot)

### Device info
**Device:** Moto X Pure
**Signal version:** 3.27.1

### Link to debug log
( Can post this if necessary, but a little uncomfortable with posting the raw debug log atm )

**Comment by 2-4601** [comment]
Seems like a duplicate of #5945.

**Comment by 6lancmange** [comment]
No, this is very clearly a completely different issue. That one has a problem with notifications coming with every (fresh) message in close succession. This is about receiving notifications for messages that have already been read on a different client. They shouldn't be treated as new messages at all.
